### DISCUSSION TITLE
OpenGraph images are not working, now I want to disable 'assets.tina.io'

### DISCUSSION THREAD

**Discussion by simnalamburt** [tinacms/tinacms#4428] [Help] [answered]
I tried to set up OpenGraph and Twitter cards on a webpage created with TinaCMS, and discovered that Facebook doesn't have access to assets.tina.io. Twitter is also not displaying the cards properly, probably for the same reason.

<img width="1049" alt="image" src="https://github.com/tinacms/tinacms/assets/4435445/8a9ef727-7303-4fd6-927f-11bcf7d413f3">

There are two solutions that came to mind:
1. Make assets.tina.io properly accessible by Facebook and Twitter; not actionable for me
2. Disable assets.tina.io

Does anybody know how to disable it?

**Comment by simnalamburt** [comment]
I failed with below since my webpage is not 100% static

```bash
tinacms build --local -c "next build"
```

###### References
- https://github.com/tinacms/tinacms/discussions/4035#discussioncomment-6264333

**Comment by simnalamburt** [comment, marked as answer]
Possible workaround for this issue:
```js
// Reference: https://github.com/tinacms/tinacms/discussions/4428
function noTina(input) {
  const url = new URL(input, 'https://blog.portone.io')
  if (url.hostname === 'assets.tina.io') {
    url.hostname = 'blog.portone.io'

    const paths = url.pathname.split('/')
    if (paths[1].match(/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/) != null) {
      paths[1] = 'uploads'
      url.pathname = paths.join('/')
    }
  }
  return url.href
}

export async function generateMetadata(props) {
  // ...

  return { openGraph: { images: [{ url: noTina(uploadurl) }] } }
}

```

**Comment by kldavis4** [comment]
> I tried to set up OpenGraph and Twitter cards on a webpage created with TinaCMS, and discovered that Facebook doesn't have access to assets.tina.io. Twitter is also not displaying the cards properly, probably for the same reason.
> 
> <img alt="image" width="1049" src="https://private-user-images.githubusercontent.com/4435445/301877829-8a9ef727-7303-4fd6-927f-11bcf7d413f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY4OTIwNzIsIm5iZiI6MTcwNjg5MTc3MiwicGF0aCI6Ii80NDM1NDQ1LzMwMTg3NzgyOS04YTllZjcyNy03MzAzLTRmZDYtOTI3Zi0xMWJjZjdkNDEzZjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDJUMTYzNjEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzM3NjNjMDJkMWYzZDQ0NDc0ZWE0MWJjMzNjZmY4MjFlNDg1MmQxNzU0OTYxYzIzNDkzOTQ3ZDIzM2RhZjA1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.A_zbgRLCdyNVA7o_iwi7rKRLnuJjuD2rTFHh7SICOPU">
> There are two solutions that came to mind:
> 
> 1. Make assets.tina.io properly accessible by Facebook and Twitter; not actionable for me
> 2. Disable assets.tina.io
> 
> Does anybody know how to disable it?

I just tried https://blog.portone.io/analytics_beta/ in the Meta debugger and it seems to load fine now (w/ assets.tina.io)

<img width="1184" alt="Screen Shot 2024-02-02 at 11 36 53 AM" src="https://github.com/tinacms/tinacms/assets/370955/f2e83cc1-6c3a-41b6-bd5b-8d50b3317634">

**Reply by simnalamburt** [reply]
That’s because I worked around this issue with https://github.com/tinacms/tinacms/discussions/4428#discussioncomment-8347853, it does not use assets.tina.io anymore.

![image](https://github.com/tinacms/tinacms/assets/4435445/50acd771-461c-48ce-b352-ff43cdd1cfb4)

I wonder if there’s any other more graceful way to disable assets.tina.io

**Reply by kldavis4** [reply]
I think there is probably something else at play [here](https://stackoverflow.com/questions/50044469/ogimage-keeps-giving-me-corrupted-image) but if the site is completely static you can use `tinacms build --local --skip-cloud-checks` and it will not use assets.tina.io.

**Reply by simnalamburt** [reply]
As I said in https://github.com/tinacms/tinacms/discussions/4428#discussioncomment-8347513  my usecase requires admin UI and dynamic elements so unfortunately that is not applicable in this case.

I believe the sysadmin of assets.tina.io should check for reachability from Meta & X server, but I'm not on a paid plan for Tina Cloud, so I haven't been able to report this issue.

I’ll just consider this resolved with https://github.com/tinacms/tinacms/discussions/4428#discussioncomment-8347853 this workaround. But thank you for your attention. I'm really enjoying TinaCMS :)

**Reply by kldavis4** [reply]
I don't think there is any reachability issue. If you put in the url for the asset directly, you should see that it is resolved in the Meta tool. Glad you are enjoying Tina, though :)
